### DISCUSSION TITLE
prisma upsert in Vercel skipped

### DISCUSSION THREAD

**Discussion by sunrunner4kr-simon** [prisma/prisma#6817] [Q&A] [answered]
Hi all, I'm hoping someone can help me

I have an API in my next js app hosted in Vercel, that receives a soap message.
When performed locally in dev, the message is consumed and the data upsert to my local DB

```
NOTIFICATION API: Processing POST
NOTIFICATION API ItemListed: Item ID 110066333222
NOTIFICATION API ItemListed: Item Title Harry Potter and the Half Blood Prince
NOTIFICATION API ItemListed: Item SKU 1
NOTIFICATION API ItemListed: Pictures http://i1.sandbox.ebayimg.com/03/i/00/3e/60/d7_1.JPG?set_id=8800005007
NOTIFICATION API ItemListed: Stored in DB
```

However, in Vercel it's logging the console log prior to the upsert, skipping the upsert and nothing after. Vercel shows the logging to the console log, and then nothing...no errors, no errors with prisma, or connecting to the DB. So I don't know how to work out what's wrong.

![image](https://user-images.githubusercontent.com/58920692/116211798-1b21ac00-a73c-11eb-949f-ecc04640ed99.png)

You can see the missing console log in the vercel log, that IS in the local console log

I know it's connected to the DB, because the vercel-build script in the build works and created the DB tables.

here's the function in my api: -

```node js
async function handleItemListed(soapBody) {
  console.log(
    "NOTIFICATION API ItemListed: Item ID",
    soapBody.GetItemResponse.Item.ItemID
  );
  console.log(
    "NOTIFICATION API ItemListed: Item Title",
    soapBody.GetItemResponse.Item.Title
  );
  console.log(
    "NOTIFICATION API ItemListed: Item SKU",
    soapBody.GetItemResponse.Item.Quantity
  );
  console.log(
    "NOTIFICATION API ItemListed: Pictures",
    soapBody.GetItemResponse.Item.PictureDetails.PictureURL
  );

  //TODO: trigger full eBay API request to get listing details?
  const result = await prisma.listing.upsert({
    where: {
      itemId: soapBody.GetItemResponse.Item.ItemID,
    },
    update: {
      itemTitle: soapBody.GetItemResponse.Item.Title,
      quantity: soapBody.GetItemResponse.Item.Quantity,
      pictureUrl: soapBody.GetItemResponse.Item.PictureDetails.PictureURL,
    },
    create: {
      itemId: soapBody.GetItemResponse.Item.ItemID,
      itemTitle: soapBody.GetItemResponse.Item.Title,
      quantity: soapBody.GetItemResponse.Item.Quantity,
      pictureUrl: soapBody.GetItemResponse.Item.PictureDetails.PictureURL,
    },
  });

  if (result) {
    console.log("NOTIFICATION API ItemListed:", "Stored in DB");
  } else {
    console.log("NOTIFICATION API ItemListed:", "Failed to store in DB");
  }
}
```

**Comment by janpio** [comment]
That is weird. I would log the object before giving it to the `upsert` to check it contains what you expect, then there are multiple ways to add logging for Prisma (check the docs) to possibly learn a bit more where exactly it stops doing things.

**Reply by sunrunner4kr-simon** [reply]
Thanks @janpio I'll have a look at the logging

**Reply by sunrunner4kr-simon** [reply]
Well...I've added this logging:

const prisma = new PrismaClient({
  log: [
    {
      emit: 'stdout',
      level: 'query',
    },
    {
      emit: 'stdout',
      level: 'error',
    },
    {
      emit: 'stdout',
      level: 'info',
    },
    {
      emit: 'stdout',
      level: 'warn',
    },
  ],
})

and not getting any prisma logging output when hitting that function, I don't think it's even trying the prisma call

and I know the object is ok, because i've logged all the values that i'm using in the upsert

![image](https://user-images.githubusercontent.com/58920692/116268110-ea119d80-a774-11eb-92d3-27b4ecff2919.png)

**Comment by sunrunner4kr-simon** [comment, marked as answer]
This was a problem with promises etc...

Here's my working code: -

```node js
function updateAuction(soap) {
    console.log(
      "NOTIFICATION API AuctionCheckoutComplete: Item ID",
      soap.Item.ItemID
    );
    console.log(
      "NOTIFICATION API AuctionCheckoutComplete: Item Title",
      soap.Item.Title
    );

    if (soap.TransactionArray.Transaction.ContainingOrder) {
      console.log(
        "NOTIFICATION API AuctionCheckoutComplete: ",
        soap.TransactionArray.Transaction.ContainingOrder.OrderID
      );

      console.log(
        "NOTIFICATION API AuctionCheckoutComplete: ",
        soap.TransactionArray.Transaction.ContainingOrder.OrderStatus
      );

      console.log(
        "NOTIFICATION API AuctionCheckoutComplete: ",
        soap.TransactionArray.Transaction.ContainingOrder.MonetaryDetails
          .Payments.Payment.PaymentStatus
      );
    }

    console.log(
      "NOTIFICATION API AuctionCheckoutComplete: ",
      soap.Item.PaymentMethods
    );

    return prisma.order.upsert({
      where: {
        orderId: soap.TransactionArray.Transaction.ContainingOrder.OrderID,
      },
      update: {
        amountPaid: soap.TransactionArray.Transaction.AmountPaid._,
        buyerName:
          soap.TransactionArray.Transaction.Buyer.BuyerInfo.ShippingAddress
            .Name,
      },
      create: {
        orderId: soap.TransactionArray.Transaction.ContainingOrder.OrderID,
        amountPaid: soap.TransactionArray.Transaction.AmountPaid._,
        buyerName:
          soap.TransactionArray.Transaction.Buyer.BuyerInfo.ShippingAddress
            .Name,
      },
    });
  }
```

```node js
updateAuction(soapBody.GetItemTransactionsResponse)
                  .then((response) => {
                    console.log("AuctionCheckoutComplete received: ", response);
                    res.status(200);
                    res.end();
                  })
                  .catch((error) => {
                    res.status(500);
                    console.log(error);
                  });
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
